fileshare/event_manager.go

```diff
@@ -66,17 +70,60 @@ func NewEventManager(
 	filesystem Filesystem,
 	defaultDownloadDir string,
 ) *EventManager {
-	return &EventManager{
+	em := &EventManager{
 		isProd:                isProd,
 		liveTransfers:         map[string]*LiveTransfer{},
 		transferSubscriptions: map[string]chan TransferProgressInfo{},
 		meshClient:            meshClient,
 		osInfo:                osInfo,
 		filesystem:            filesystem,
 		defaultDownloadDir:    defaultDownloadDir,
+		syncEvents:            make(chan []Event),
+		syncDoneCh:            make(chan struct{}),
+		asyncEvents:           make(chan []Event, 2048),
+	}
+	go em.process()
+
+	return em
+}
+
+func (em *EventManager) process() {
+	fn := func(ev []Event) {
+		em.mutex.Lock()
+		for _, e := range ev {
+			em.handleEvent(e)
+		}
+		em.mutex.Unlock()
+	}
+
+	for {
+		select {
+		case e, ok := <-em.asyncEvents:
+			if !ok {
+				log.Println(internal.WarningPrefix + " asyncEvents channel closed")
```

Comment:
Copilot actually suggested me this, and I guess it's because of this: 
https://github.com/NordSecurity/nordvpn-linux/blob/0e0e77eb68695f98469ee6dfe8dc21478d923ebc/fileshare/event_manager.go#L179

Will fix